sn_node/src/comm/mod.rs

```diff
@@ -353,66 +251,69 @@ impl Comm {
             })
             .collect();
 
-        let delivery_group_size = recipients.len();
-        let mut next = delivery_group_size;
-        let mut successes = 0;
-        let mut failed_recipients = vec![];
-
-        let mut try_next = |error, recipient, tasks: &mut FuturesUnordered<_>| {
-            warn!("during sending, received error {:?}", error);
-            failed_recipients.push(recipient);
+        let watchers = futures::future::join_all(tasks).await;
 
-            if next < recipients.len() {
-                let mut msg = wire_msg.clone();
-                msg.set_dst_xorname(recipients[next].name());
-                tasks.push(self.send_to_one(recipients[next], msg, priority));
-                next += 1;
-            }
-        };
+        for (recipient, result) in watchers {
+            let name = recipient.name();
+            let addr = recipient.addr();
 
-        while let Some((recipient, result)) = tasks.next().await {
             match result {
                 Err(error) => {
                     // there is only one type of error returned: [`Error::InvalidState`]
                     // which should not happen (be reachable) if we only access PeerSession from Comm
                     // The error means we accessed a peer that we disconnected from.
                     // So, this would potentially be a bug!
-                    //
-                    // (let's log that bug here, but continue running anyway, as it isn't that critical)
                     warn!(
                         "Accessed a disconnected peer: {}. This is potentially a bug!",
                         recipient
                     );
-                    try_next(error, recipient, &mut tasks);
+                    error!(
+                        "Sending message (msg_id: {:?}) to {:?} (name {:?}) failed as we have disconnected from the peer. (Error is: {})",
+                        msg_id,
+                        addr,
+                        name,
+                        error,
+                    );
```

Comment:
These log lines can be compressed down to ~1-2 lines by inlining the variables:
```suggestion
                    error!(
                        "Sending message (msg_id: {msg_id:?}) to {addr:?} (name {name:?}) failed as we have disconnected from the peer. (Error is: {error})"
                    );
```